src/index.js

```diff
@@ -1,13 +1,36 @@
 // @flow
 
 import type {
+  NavigationAction,
+  NavigationContainer,
   NavigationEventCallback,
   NavigationState,
 } from 'react-navigation';
-import type { Middleware } from 'redux';
+import type { Middleware, Reducer } from 'redux';
 
+import { NavigationActions } from "react-navigation";
 import invariant from 'invariant';
 
+// @TODO ask if this is the correct type
+type Navigator = NavigationContainer<*, *, *>;
+
+function createNavigationReducer(navigator: Navigator): Reducer<*,*> {
+  const initialState = navigator.router.getStateForAction(
+    NavigationActions.init(),
+    null
+  );
+
+  return (
+    // @TODO find the function needed
+    // getStateForAction it giving undefined back
+    state: NavigationState = initialState,
+    action: NavigationAction
+  ): NavigationState => {
+    const newState = navigator.router.getStateForAction(action, state);
```

Comment:
Does `getStateForAction` actually return null ever? In what circumstances does this happen?

Reply:
@Ashoat No sure mate, I am using it as it is in a project and never give me `null` back but probably the type checking is wrong.

I am not sure

Reply:
Please read the code and try to figure out! If it can't be null, then update the libdef correspondingly. Note that `getStateForAction` is defined in `TabRouter` and `StackRouter`

Reply:
@Ashoat yes it does https://github.com/react-navigation/react-navigation/blob/2bb91a6740e2f1ade012906dc03c972ba6375cc3/src/routers/TabRouter.js#L102

So far, this is the only place so StackRouter is safe.

Maybe we should request the initial state instead of calling this function? 

So far I dont have problem at all with this method but let me know.

Reply:
@Ashoat from the Redux integration documentation

```

const initialState = AppNavigator.router.getStateForAction(AppNavigator.router.getActionForPathAndParams('Login'));
```

Technically this is exactly what I am doing in terms of it could be `null` so.